tiledb/sm/array_schema/test/unit_domain.cc

```diff
@@ -128,3 +130,41 @@ TEST_CASE("Domain: Test deserialization", "[domain][deserialize]") {
   CHECK(dim2->cell_val_num() == cell_val_num2);
   CHECK(dim2->filters().size() == num_filters2);
 }
+
+TEST_CASE("Domain: Cells per tile larger than dimension type", "[domain]") {
```

Comment:
This is testing the change properly. However, I'm worried there is something else we might have missed somewhere. I would feel much more comfortable if we would write a dense tile with this configuration with all unique values and read it back.

Reply:
As discussed offline, I am handing the testing off to you to verify fixing this issue and allowing the write doesn't lead to bad data being written to a tile.